tests/buffering.rs

```diff
@@ -67,21 +67,14 @@ fn test_buffering() {
         .send_all(input_events_stream);
     let _ = rt.block_on(send).unwrap();
 
-    // A race caused by `rt.block_on(send).unwrap()` is handled here. For some
-    // reason, at times less events than were sent actually arrive to the
-    // `source`.
-    // We mitigate that by waiting on the event counter provided by our source
-    // mock.
-    test_util::wait_for_atomic_usize(source_event_counter, |x| x == num_events);
```

Comment:
Instead of removing this altogether, can we switch from `test_util::wait_for_atomic_usize` to an `assert!(x == num_events)`? This would be great help to spot the problem in case it comes back again.

Reply:
True.